akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -2223,8 +2233,17 @@ trait FlowOps[+Out, +Mat] {
    *
    * @see [[#splitWhen]]
    */
-  def splitWhen(p: Out => Boolean): SubFlow[Out, Mat, Repr, Closed] =
-    splitWhen(SubstreamCancelStrategy.drain)(p)
+  def splitWhen(p: Out => Boolean): SubFlow[Out, Mat, Repr, Closed] = {
+    val merge = new SubFlowImpl.MergeBack[Out, Repr] {
+      override def apply[T](flow: Flow[Out, T, NotUsed], breadth: Int): Repr[T] =
+        via(Split.when(p)).map(_.via(flow)).via(new FlattenMerge(breadth))
+    }
+
+    val finish: Sink[Out, NotUsed] => Closed = s =>
+      via(Split.when(p)).to(Sink.foreach(_.runWith(s)(GraphInterpreter.currentInterpreter.materializer)))
+
+    new SubFlowImpl(Flow[Out], merge, finish)
```

Comment:
This seems repeated quite a bit more now than in the original code, could it not delegate to a single method just like before? It would be fine with an internal/private method if the strategy needs to be passed in.

Reply:
Sure will do